src/index/LocalVocabEntry.cpp

```diff
@@ -14,12 +14,20 @@ auto LocalVocabEntry::positionInVocabExpensiveCase() const -> PositionInVocab {
   // this word would be stored if it were present.
   const IndexImpl& index = IndexImpl::staticGlobalSingletonIndex();
   PositionInVocab positionInVocab;
+
   const auto& vocab = index.getVocab();
 
-  auto [lower, upper] = vocab.getPositionOfWord(toStringRepresentation());
-  AD_CORRECTNESS_CHECK(upper.get() - lower.get() <= 1);
-  positionInVocab.lowerBound_ = lower;
-  positionInVocab.upperBound_ = upper;
+  auto [lower, upper] = [&]() {
+    if (auto opt = index.encodedValueManager().encode(toStringRepresentation());
+        opt.has_value()) {
+      return std::pair{opt.value(), Id::fromBits(opt.value().getBits() + 1)};
```

Comment:
I can't quite follow the logic here. I get why `VocabIndex` was replaced by the new placeholder type. But why does this work? These upper and lower bounds don't seem to have anything to do with the vocab index the else case provides?